redfish-core/lib/log_services.hpp

```diff
@@ -5368,4 +5419,1084 @@ inline void requestRoutesPostCodesEntry(App& app)
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        if (objType[0].first.empty())
+        {
+            BMCWEB_LOG_ERROR << "The retrieved HardwareIsolation dbus "
+                                "name is empty";
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        // Delete all HardwareIsolation entries
+        crow::connections::systemBus->async_method_call(
+            [asyncResp](const boost::system::error_code ec1,
+                        const sdbusplus::message::message& msg) {
+            if (!ec1)
+            {
+                messages::success(asyncResp->res);
+                return;
+            }
+
+            BMCWEB_LOG_ERROR << "DBUS response error [" << ec1.value() << " : "
+                             << ec1.message()
+                             << "] when tried to delete all HardwareIsolation "
+                                "entries";
+
+            const sd_bus_error* dbusError = msg.get_error();
+
+            if (dbusError == nullptr)
+            {
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            BMCWEB_LOG_ERROR << "DBus ErrorName: " << dbusError->name
+                             << " ErrorMsg: " << dbusError->message;
+
+            if (strcmp(dbusError->name, "xyz.openbmc_project.Common.Error."
+                                        "NotAllowed") == 0)
+            {
+                messages::chassisPowerStateOffRequired(asyncResp->res,
+                                                       "chassis");
+            }
+            else
+            {
+                BMCWEB_LOG_ERROR << "DBus Error is unsupported so "
+                                    "returning as Internal Error";
+                messages::internalError(asyncResp->res);
+            }
+            return;
+            },
+            objType[0].first, "/xyz/openbmc_project/hardware_isolation",
+            "xyz.openbmc_project.Collection.DeleteAll", "DeleteAll");
+        },
+        "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetObject",
+        "/xyz/openbmc_project/hardware_isolation",
+        std::array<const char*, 1>{"xyz.openbmc_project.Collection.DeleteAll"});
+}
+
+/**
+ * @brief API used to route the handler for HardwareIsolation Redfish
+ *        LogServices URI
+ *
+ * @param[in] app - Crow app on which Redfish will initialize
+ *
+ * @return The appropriate redfish response for the given redfish request.
+ */
+inline void requestRoutesSystemHardwareIsolationLogService(App& app)
+{
+    BMCWEB_ROUTE(app,
+                 "/redfish/v1/Systems/system/LogServices/HardwareIsolation/")
+        .privileges(redfish::privileges::getLogService)
+        .methods(boost::beast::http::verb::get)(
+            getSystemHardwareIsolationLogService);
+
```

Comment:
New bmcweb code upstream also has HEAD. 
Oh well.

Reply:
I am unable to follow the comment. Could you please elaborate it.
Thank you!